File: 2023/day1/day1_ex1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>

using namespace std;

int main () {
    ifstream myfile("day1_input.txt");
    string input;
    int sum = 0;
    string nbr;


    if (myfile.is_open()) {
        // while (myfile) {
            while(getline (myfile, input)) {
                nbr = "";
                for (int i = 0; i < input.length(); i ++) {
                    if (isdigit(input[i]))
                        nbr.push_back(input[i]);
                }
                if (nbr.length() == 1)
                    sum += stoi(nbr + nbr);
                else if ((nbr.length() == 2))
                    sum += stoi(nbr);
                else if (nbr.length() > 2)
                    sum += stoi(string(1, nbr[0]) + string(1, nbr[nbr.length() - 1]));
            }
        // }
    }
    cout << sum << endl;
}

File: 2023/day1/day1_ex1.py
def main():
    try:
        f = open('day1_input.txt')
        sum = 0
        lines = f.readlines()
        for line in lines:
            num = [int(x) for x in [*line] if x.isdigit()]
            # num = []
            # for x in [*line]:
            #     if x.isdigit():
            #         num.append(int(x))
            # print(num)
            if len(num) == 1:
                sum += num[0] * 10 + num[0]
            else:
                sum += num[0] * 10 + num[len(num) - 1]
        print(sum)
        f.close()
    except AssertionError as e:
        print(AssertionError.__name__, e, sep=": ")

if __name__ == "__main__":
    main()